src/cpp/fastdds/subscriber/DataReaderImpl.cpp

```diff
@@ -1743,8 +1743,9 @@ void DataReaderImpl::release_payload_pool()
     if (!is_custom_payload_pool_)
     {
         PoolConfig config = PoolConfig::from_history_attributes(history_.m_att);
+        std::shared_ptr<fastrtps::rtps::ITopicPayloadPool> topic_payload_pool_;
+        topic_payload_pool_ = std::dynamic_pointer_cast<fastrtps::rtps::ITopicPayloadPool>(payload_pool_);
         topic_payload_pool_->release_history(config, true);
```

Comment:
This is in fact giving a warning in Windows
```suggestion
        std::shared_ptr<fastrtps::rtps::ITopicPayloadPool> topic_payload_pool = std::dynamic_pointer_cast<fastrtps::rtps::ITopicPayloadPool>(payload_pool_);
        topic_payload_pool->release_history(config, true);
```